e2e/testcases/cli_test.go

```diff
@@ -1068,3 +1068,22 @@ func TestNomosVersion(t *testing.T) {
 		nt.T.Fatalf("Expected to find config-sync component in output:\n%s\n", string(out))
 	}
 }
+
+func TestNomosStatusFilter(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.NomosCLI)
+
+	nt.MustKubectl("create", "namespace", "bookinfo")
+	nt.MustKubectl("apply", "-f", "../testdata/reconciler-manager/reposync-sample.yaml")
+
+	// get status
+	cmd := nt.Command("nomos", "status", "--rsyncname", "repo-sync")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		nt.T.Log(string(out))
+		nt.T.Fatal(err)
+	}
+
+	if strings.Contains(string(out), "root-sync") {
```

Comment:
This assertion isn't very strong. It should validate more about the output, e.g. that other syncs are not present in the output.

Perhaps we could mask the dynamic parts of the output (I assume commit SHA) so that we could do a more explicit assertion?

Reply:
i have updated the assertion to validate namespace/resource that we expect to see and should not see in the result. Also added more test for different use case